packages/compass-e2e-tests/installers/helpers.ts

```diff
@@ -0,0 +1,19 @@
+import type { SpawnSyncReturns } from 'child_process';
+
+export function assertSpawnSyncResult(
```

Comment:
Non-blocking, and no need to change anything in this PR, but:

The typical best practice in Node.js in general is to use `execFile` (and usually promisified so it's not completely unworkable). `spawnSync` is obviously also _fine_ to use out of convenience in tests, since there's no need to care about performance or limitations of the synchronous method as much, but I'd be wary of building an entire test helper system around it – if you're at the point where you're introducing helpers already, you might as well just stick to the async method. (Again, this all matters only to a limited extent, but "follow best practices for production code even in tests" just makes it easier to also follow best practices in actual production code)

Reply:
Regarding sync vs async: I was on the fence and I'll just change it to async.

Regarding exec vs execFile vs spawn vs spawn: I have no idea when to use which. I suppose exec() at least has a callback for when it finishes, but I got stuck on the fact that in a general sense I don't want all the stdout and stderr in a variable. In these cases there's little output, but in a more general sense you can never be too sure. To me the ideal seems to pass it through to stdout and stderr and just resolve a promise with an error. None of them seem to do that or close to it out of the box.

Reply:
ie. I basically want to happen what happens when you call a program in a shell script 😆

Reply:
I made a promisified async spawn helper rather and removed assertSpawnSyncResult. This way I can error on anything except exit code 0 and stdio gets passed through rather than accumulate all the e2e test output in the result.